Upstream-Status: Backport

Part 3/3 of making the GN binary link correctly. More information:
https://crbug.com/934793

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
From e15f243a7af794b1bdc915885717749af853a3bb Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Mon, 25 Feb 2019 15:09:52 +0100
Subject: [PATCH] gen.py: Pass -pthread, not -lpthread, in ldflags.

This was shamefully overlooked in 1152229e ("gen.py: Pass -pthread via
ldflags, not libs"), where I was supposed to revert the entirety of commit
082fbe39 ("Pass -lpthread when linking instead of -pthread") but did not
notice that 0ac93dd2 ("For proper order put -lpthreads as a 'library' not a
'linker flag'") continued passing -lpthread instead of -pthread.

Bug: chromium:934793
Change-Id: I1fa6d2c8b32a86d621a74ae97f391fbe4ea5e989
Reviewed-on: https://gn-review.googlesource.com/c/4100
Reviewed-by: Nico Weber <thakis@chromium.org>
Commit-Queue: Nico Weber <thakis@chromium.org>
---
 build/gen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/gen.py b/build/gen.py
index c426c0bf..a7142fab 100755
--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -334,7 +334,7 @@ def WriteGNNinja(path, platform, host, options):
       ldflags.append('-maix64')
 
     if platform.is_posix():
-      ldflags.append('-lpthread')
+      ldflags.append('-pthread')
 
     if options.use_lto:
       cflags.extend(['-flto', '-fwhole-program-vtables'])
-- 
2.20.1

